packages/flutter_tools/lib/src/plugins.dart

```diff
@@ -250,11 +275,13 @@ Plugin _pluginFromPubspec(String name, Uri packageRoot) {
     return null;
   }
   final String packageRootPath = fs.path.fromUri(packageRoot);
+  final YamlMap dependencies = pubspec['dependencies'];
```

Comment:
This will capture direct dependencies, but not transitive dependencies - is this an issue?. Consider:

plugin A depends on package 1
package 1 depends on plugin B

.flutter-plugins will contain plugin A/plugin B while the dependencies file will not link them.

Reply:
From offline discussion, this isn't a supported use case - and that platform code cannot depend on eachother. Therefore it doesn't seem like the gradle task ordering needs to take this into consideration